src/components/TableOfContents.tsx

```diff
@@ -416,52 +335,96 @@ const TableOfContents: React.FC<IProps> = ({
-            <ButtonLink to={editPath} variant="outline" hideArrow mt={0}>
-              <ButtonContent>
-                <GithubIcon name="github" />{" "}
-                <span>
+    // TODO: Switch to `above="lg"` after completion of Chakra Migration
+    <Show breakpoint="(min-width: 1025px)">
+      <Box
+        as="aside"
+        position="sticky"
+        top="7.25rem" // Account for navbar
+        p={4}
+        pe={0}
+        maxW="25%"
+        minW={48}
+        height={calc.subtract("100vh", "80px")}
+        overflowY="auto"
+        {...rest}
+      >
+        <List {...outerListProps}>
+          {!hideEditButton && (
+            <ListItem mb={2}>
+              <ButtonLink to={editPath} variant="outline" hideArrow mt={0}>
+                <Flex alignItems="center">
+                  <Icon as={FaGithub} color="text" boxSize={6} me={2} />
                   <Translation id="edit-page" />
-                </span>
-              </ButtonContent>
-            </ButtonLink>
-          </ButtonContainer>
-        )}
-        {shouldShowZenModeToggle && (
-          <ZenModeContainer>
-            <ZenModeText>
-              <Translation id="zen-mode" />
-            </ZenModeText>
-            <ZenModeToggleContainer onClick={(_e) => handleZenModeChange()}>
-              <Icon name={isZenMode ? "toggleOn" : "toggleOff"} size="1.6rem" />
-            </ZenModeToggleContainer>
-          </ZenModeContainer>
-        )}
-        <Header>
-          <Translation id="on-this-page" />
-        </Header>
-        <ItemsList
-          items={items}
-          depth={0}
-          maxDepth={maxDepth ? maxDepth : 1}
-          activeHash={activeHash}
-        />
-      </OuterList>
-    </Aside>
+                </Flex>
+              </ButtonLink>
+            </ListItem>
+          )}
+          {shouldShowZenModeToggle && (
+            <Flex
+              as={ListItem}
+              alignItems="center"
+              mb={2}
+              py="2px"
+              opacity={0.8}
+              fontSize="sm"
+            >
+              <Box as="span" me={2} id="zen-mode">
+                <Translation id="zen-mode" />
+              </Box>
+              <Switch
+                aria-labelledby="zen-mode"
+                sx={{
+                  "& .chakra-switch__track": {
+                    background: "transparent",
+                    border: "2px solid",
+                    borderColor: "secondary",
+                    p: 0,
+                    "&[data-checked]": {
+                      background: "secondary",
+                    },
+                  },
+                  "& .chakra-switch__thumb": {
+                    background: "ednBackground",
+                    outline: "2px solid",
+                    outlineColor: "secondary",
+                  },
+                }}
+                isChecked={isZenMode}
+                onChange={() => handleZenModeChange()}
+              />
```

Comment:
Replace toggle icons in a `span` element with the Chakra [Switch](https://chakra-ui.com/docs/components/switch/usage) component for improved accessibility with a checkbox-like toggle. Creates a different sized element, readjusts space around it, and includes custom styling to closely mimic the original icons.